test/Streamly/Test/Prelude/Serial.hs

```diff
@@ -106,55 +107,160 @@ splitOnSuffixSeq = do
         S.toList
              $ IS.splitOnSuffixSeq (A.fromList pat) (FL.toList) (S.fromList xs)
 
+seqSplitterProperties ::
+       forall a. (Arbitrary a, Eq a, Show a, Storable a, Enum a)
+    => a
+    -> String
+    -> Spec
+seqSplitterProperties sep desc = do
+    describe (desc <> " splitOnSeq")
+        $ do
+
+            forM_ [0, 1, 2, 4] $ intercalateSplitEqId
+            forM_ [0, 1, 2, 4]
+                $ concatSplitIntercalateEqConcat
+                      splitOnSeq_
+                      intercalate
+                      IS.intercalate
+
+            -- Exclusive case
+            splitIntercalateEqId splitOnSeq_ intercalate IS.intercalate
+
+    describe (desc <> " splitOnSuffixSeq")
+        $ do
+
+            forM_ [0, 1, 2, 4] $ intercalateSplitEqIdNoSepEnd
+            forM_ [0, 1, 2, 4]
+                $ concatSplitIntercalateEqConcat
+                      splitOnSuffixSeq_
+                      intercalateSuffix
+                      IS.intercalateSuffix
+
+            -- Exclusive case
+            splitIntercalateEqId
+                splitOnSuffixSeq_
+                intercalateSuffix
+                IS.intercalateSuffix
+
+    where
```

Comment:
This where clause has become too big. We could possibly float these definitions out to the top level. We can also split the tests for `splitOnSeq` and `splitOnSuffixSeq` in separate functions.

Reply:
These functions are pretty specific to `splitOnSeq`. I don't see any need for moving them out. I did initially split the tests but that led to some code duplication earlier